flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -246,52 +254,83 @@ public Builder uidPrefix(String newPrefix) {
         }
       }
 
-      // Find out the equality field id list based on the user-provided equality field column names.
-      List<Integer> equalityFieldIds = Lists.newArrayList();
-      if (equalityFieldColumns != null && equalityFieldColumns.size() > 0) {
-        for (String column : equalityFieldColumns) {
-          org.apache.iceberg.types.Types.NestedField field = table.schema().findField(column);
-          Preconditions.checkNotNull(field, "Missing required equality field column '%s' in table schema %s",
-              column, table.schema());
-          equalityFieldIds.add(field.fieldId());
+      // set name and uid for mappedRowDataInput if needed
+      if (mappedRowDataInput != null) {
+        if (uidPrefix != null) {
+          rowDataInput = mappedRowDataInput
+              .name(operatorName(uidPrefix))
+              .uid(uidPrefix + "-mapper");
+        } else {
+          rowDataInput = mappedRowDataInput;
         }
       }
 
       // Convert the requested flink table schema to flink row type.
       RowType flinkRowType = toFlinkRowType(table.schema(), tableSchema);
 
       // Distribute the records from input data stream based on the write.distribution-mode.
-      rowDataInput = distributeDataStream(rowDataInput, table.properties(), table.spec(), table.schema(), flinkRowType);
+      DataStream<RowData> distributeStream = distributeDataStream(
+          rowDataInput, table.properties(), table.spec(), table.schema(), flinkRowType);
```

Comment:
Should we also add the uid & name if we've already chained the `PartitionKeySelector` ?

Reply:
`keyBy` returns a `KeyedStream` which extends from `DataStream` that doesn't expose `name` and `uid`